src/v/cluster/scheduling/constraints.h

```diff
@@ -27,8 +28,21 @@ hard_constraint_evaluator on_nodes(const std::vector<model::node_id>&);
 hard_constraint_evaluator
 distinct_from(const std::vector<model::broker_shard>&);
 
+hard_constraint_evaluator disk_not_overflowed_by_partition(
+  const double max_disk_usage,
+  const size_t partition_size,
+  const std::unordered_map<model::node_id, uint64_t>&
+    assigned_reallocation_sizes,
+  const std::unordered_map<model::node_id, node_health_report>& health_reports);
+
 soft_constraint_evaluator least_allocated();
 
+soft_constraint_evaluator least_disk_filled(
+  const double max_disk_usage,
+  const std::unordered_map<model::node_id, uint64_t>&
+    assigned_reallocation_sizes,
+  const std::unordered_map<model::node_id, node_health_report>& health_reports);
+
```

Comment:
Can we add short comments to those method describing the arguments. Also maybe it would be more efficient to use `absl::flat_hash_map` for `node_id` keyed maps ?